### ISSUE TITLE
Do not support float input in learning rate warmup

### ISSUE THREAD

**Issue by Officium** [PaddlePaddle/Paddle#18408]
paddle 1.4.1 CPU
Just adapt from the document.
```python
import paddle.fluid as fluid

warmup_steps = 50
start_lr = 1. / 3.
end_lr = 0.1
decayed_lr = fluid.layers.linear_lr_warmup(
    0.2,
    warmup_steps, start_lr, end_lr)
```

**Comment by qingqing01** [comment]
@Officium 

Thanks for point this.

Yeah, do not support set float now.  You can use it like:


```python
import paddle.fluid as fluid
import numpy as np

warmup_steps = 50
start_lr = 1. / 3.
end_lr = 0.1
decayed_lr = fluid.layers.linear_lr_warmup(
    np.array(0.2).astype('float32'),
    warmup_steps, start_lr, end_lr)
print(decayed_lr)
```


I wil enhance it.

**Comment by Officium** [comment]
@qingqing01 Thanks for your response. There are more similar issues. I list two examples that are easy to reproduce.
1. int input to learning_rate
```python
fluid.optimizer.AdamOptimizer(learning_rate=1)
```
2. int input to regularizer (it is usually commonly used when passing parameter in cmd.)
```python
import numpy as np
import paddle.fluid as fluid


dim = 5
x = fluid.layers.data(name='x', shape=[dim], dtype='float32')
predict = fluid.layers.fc(x, size=2)
loss = fluid.layers.mean(predict)
optimizer = fluid.optimizer.AdamOptimizer(
    learning_rate=0.1,
    regularization=fluid.regularizer.L2DecayRegularizer(0)
)
optimizer.minimize(loss)

place = fluid.CPUPlace()
exe = fluid.Executor(place)
exe.run(fluid.default_startup_program())
x = np.random.random((2, dim)).astype('float32')
res = exe.run(fluid.default_main_program(),
              feed={'x': x},
              fetch_list=[loss])

```

Please fix them altogether.

**Comment by Officium** [comment]
@wangguibao @qingqing01 Already done? Where is the PR?

**Comment by wangguibao** [comment]
还没修呢呀，您要是着急就用上面青青提的方法呗
